manager.py

```diff
@@ -87,10 +89,11 @@ def configure(self, context):
 
         # fetch instance specific settings
         default = Persistence().load("api_key", default=None)
-        self.api_key = NIOEnvironment.get_variable('API_KEY', 
-                                                   default=default)
+        self._api_key = NIOEnvironment.get_variable('API_KEY',
+                                                    default=default)
+        self._instance_id = NIOEnvironment.get_variable('INSTANCE_ID')
```

Comment:
I don't think either of these (instance id and api key) should be environment variables in the code, we should pull them from the configuration file. If the instance wants to use environment variables they can set that in the conf file. That should be the default behavior though.

I think we should have two configuration settings for this:
```ini
configuration_fetch_endpoint: https://api.n.io/v1/instances/[[INSTANCE_ID]]/configuration
instance_api_key: [[API_KEY]]
```

Reply:
Agree, I've noticed what is becoming like a pattern which is to first read from settings to get a default value, and then read from persistence using this default as a fallback, this will apply to `instance_api_key` and `instance_id`

Regarding `configuration_fetch_endpoint` we've been using a more general `config_api_url_prefix` setting, to it we normally append the particulars of the request either get config versions, actual configuration, etc.